plugins/kn-migration/cmd/client.go

```diff
@@ -0,0 +1,235 @@
+// Copyright © 2019 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"github.com/fatih/color"
+	api_errors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	api_serving "knative.dev/serving/pkg/apis/serving"
+	serving_v1_api "knative.dev/serving/pkg/apis/serving/v1"
+	serving_v1_client "knative.dev/serving/pkg/client/clientset/versioned/typed/serving/v1"
+)
+
+type MigrationClient interface {
+	// Create service struct from provided options
+	ConstructService(originalservice serving_v1_api.Service) *serving_v1_api.Service
+
+	// Create revision struct from provided options
+	ConstructRevision(originalrevision serving_v1_api.Revision, config_uuid types.UID) *serving_v1_api.Revision
+
+	// Check if service exists
+	ServiceExists(name string) (bool, error)
+
+	// Get a config by service name
+	GetConfig(name string) (*serving_v1_api.Configuration, error)
+
+	// Get a service by name
+	GetService(name string) (*serving_v1_api.Service, error)
+
+	// Get service list
+	ListService() (*serving_v1_api.ServiceList, error)
+
+	// Create a service
+	CreateService(service *serving_v1_api.Service) (*serving_v1_api.Service, error)
+
+	// Delete a service by name
+	DeleteService(name string) error
+
+	// Get a revision by service name
+	GetRevision(name string) (*serving_v1_api.Revision, error)
+
+	// Create a revision
+	CreateRevision(revision *serving_v1_api.Revision, config_uuid types.UID) (*serving_v1_api.Revision, error)
+
+	// Update the given revision
+	UpdateRevision(revision *serving_v1_api.Revision) error
+
+	// Get revision list by service
+	ListRevisionByService(name string) (*serving_v1_api.RevisionList, error)
+
+	// Get service list with revisions
+	PrintServiceWithRevisions(clustername string) error
+}
+
+type migrationClient struct {
+	client    serving_v1_client.ServingV1Interface
+	namespace string
+}
+
+// Create a new client facade for the provided cl.namespace
```

Comment:
Be consistent for Godoc by adding the name of the function as the first word in comment for public functions. Use /lint if you want PROW to flag for you.

Reply:
fixed in the second commit: https://github.com/knative/client-contrib/pull/6/commits/e77b98b6cebdccc7a81fa81f9c508488611a2d3b